src/core/search/id_search_res.rs

```diff
@@ -1,6 +1,7 @@
 use crate::core::lookup::lookup_table::LookupTableLevel;
 use crate::core::Identifier;
 
+#[derive(Debug)]
 pub struct IdentifierSearchResult {
     target: Identifier,
     level: LookupTableLevel,
```

Comment:
[suggestion] Would that make sense to rename `level` to `result_level`, or something that indicates that this is the level of the found identifier, not the level used in the search request. I'll leave it up to you though, totally optional.